compiler/z/env/OMRCPU.cpp

```diff
@@ -31,6 +31,36 @@
 #include "control/Options_inlines.hpp"
 #include "env/CPU.hpp"
 
+OMR::Z::CPU::CPU(const OMRProcessorDesc& processorDescription) : OMR::CPU(processorDescription)
```

Comment:
`_supportedArch` is part of the old implementation and should be initialized else where. (In OpenJ9)
Should still use `_supportedArch(z9)` in this constructor.

@dchopra001

Reply:
Also we don't need to worry about the processor detection assertion test being triggered as currently standalone OMR doesn't use the new implementation (as its omrPortLib pointer is NULL).

Reply:
Then where should `_supportedArch` be initialized? Do we later go and update `_supportedArch`? For context, the issue here happens when the following code is run:

```
#if defined(J9VM_OPT_JITSERVER)
   // In JITServer, we would like to use JITClient's processor info for the compilation
   // The following code effectively replaces the cpu with client's cpu through the getProcessorDescription() that has JITServer support
   if (self()->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER)
      {
      OMRProcessorDesc JITClientProcessorDesc = TR::Compiler->target.cpu.getProcessorDescription();
      _target.cpu = TR::CPU(JITClientProcessorDesc);
      }
   else
#endif /* defined(J9VM_OPT_JITSERVER) */
```

The constructor that I've modified here is invoked in the above code. And we don't always use this constructor for normal initialization.

Reply:
`_supportedArch` is initialized in `J9::Z::CPU::initializeS390ProcessorFeatures()`.
Okay I understand why you made this change now.
To make this work, I think we need to add `_target.cpu.initializeS390ProcessorFeatures()` after `_target.cpu = TR::CPU(JITClientProcessorDesc)`.